cns/internal/heartbeat.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018 Microsoft. All rights reserved.
+// MIT License
+
+package internal
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-container-networking/aitelemetry"
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/restserver"
+	"github.com/Azure/azure-container-networking/cns/types"
+)
+
+func SendHeartBeat(ctx context.Context, heartbeatIntervalInMins int, homeAzMonitor *restserver.HomeAzMonitor) {
```

Comment:
It would be a more flexible design if this accepted a `time.Duration`, and let the caller work out how to construct the right duration value. The fact that the configuration is defined in minutes shouldn't influence how this function operates. As it stands, this would be hard to unit test because we're forced to choose a minute value and stall the test suite for N minutes.